src/pybind/mgr/cephadm/serve.py

```diff
@@ -179,6 +179,7 @@ def _autotune_host_memory(self, host: str) -> None:
     def _refresh_hosts_and_daemons(self) -> None:
         bad_hosts = []
         failures = []
+        osd_metadata = self.mgr.osd_service.get_osd_metadata()
```

Comment:
@strenuous-life could you revise the title of the commit?

> _ceph_get is called many time concurrently which cause deadlock

the title of the commit message reads like a bug report. ideally, the title of the commit message should use the imperative mood. see also https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes . please use `git commit --amend`  and `git push -f` for revising the commit message and its title.

so the title could be revised like

> cache osd meta data in _refresh_hosts_and_daemons()

and you could put more details in the body of the commit message to explain the rationales behind this change.